content/docs/core/1.2.0/how-to-guides/limit-orderline-quantity.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Limit Order Line Quantity
+description: How-To Guide to limit order line quantity in Vendr, the eCommerce solution for Umbraco v8+
+---
+
+In this guide we will be looking at Validation events in Vendr to be able to limit order line quantity based on the existing stock value on the product and the existing quantities of the product in the cart.
+
+## ProductAddValidationHandler
+
+When adding a product to cart we need to verify the product is in stock and the customer not already has the remaining quantities in cart.
+
+````csharp
+public class ProductAddValidationHandler : ValidationEventHandlerBase<ValidateOrderProductAdd>
+{
+    private readonly IUmbracoContextAccessor _context;
+
+    public ProductAddValidationHandler(IUmbracoContextAccessor context)
+    {
+        _context = context;
+    }
+
+    public override void Validate(ValidateOrderProductAdd evt)
+    {
+        var order = evt.Order;
+        var productReference = evt.ProductReference;
+
+        var product = _context.UmbracoContext.Content.GetById(new Guid(productReference));
+        var stock = product.Value<decimal?>("stock");
+
+        var totalQuantities = order?.OrderLines.Where(x => x.ProductReference == productReference).Sum(x => x.Quantity) ?? 0;
+
+        if (stock.HasValue && totalQuantities >= stock.Value)
+            evt.Fail($"Only {stock} quantities can be purchased for {productReference}");
+    }
+}
+
+````
+
+## OrderLineQuantityValidationHandler
+
+Furthermore when changing order line quantity on cart page, we need to ensure to quantities being changed to are in stock.
```

Comment:
- ...changing **the** order line...
- ...ensure **the** quantities being changed are in stock.